modules/services/window-managers/fluxbox.nix

```
@@ -107,8 +107,9 @@ in {
         mkIf (cfg.windowmenu != "") { text = cfg.windowmenu; };
     };
 
-    xsession.windowManager.command = concatStringsSep " "
-      ([ "${cfg.package}/bin/fluxbox" ]
-        ++ escapeShellArgs (remove "" cfg.extraCommandLineArgs));
+    xsession.windowManager.command = let
+      exe = "${cfg.package}/bin/fluxbox";
+      args = escapeShellArgs (remove "" cfg.extraCommandLineArgs);
+    in "${exe} ${args}";
```

This should work:

```suggestion
    xsession.windowManager.command = concatStringsSep " " 
       [ "${cfg.package}/bin/fluxbox" ] 
         [ escapeShellArgs (remove "" cfg.extraCommandLineArgs) ];
```

---

This doesn't work when I try in the repl:

```
Welcome to Nix 2.15.1. Type :? for help.

nix-repl> :lf nixpkgs
Added 15 variables.

nix-repl> :lf .
Added 10 variables.

nix-repl> cfg = homeConfigurations."nixos@vmwnix".config.xsession.windowManager.fluxbox

nix-repl> let
            inherit (lib) concatStringsSep escapeShellArgs remove;
          in (
            concatStringsSep " "
            [ "${cfg.package}/bin/fluxbox" ]
            [ escapeShellArgs (remove "" cfg.extraCommandLineArgs) ]
          )
error: attempt to call something which is not a function but a string with context

       at «string»:4:3:

            3| in (
            4|   concatStringsSep " "
             |   ^
            5|   [ "${cfg.package}/bin/fluxbox" ]
```

versus

```
nix-repl> let
            inherit (lib) concatStringsSep escapeShellArgs remove;
            exe = "${cfg.package}/bin/fluxbox";
            args = escapeShellArgs (remove "" cfg.extraCommandLineArgs);
          in
            "${exe} ${args}"
"/nix/store/0cdjylqwq2rjg3ha238wmzv50avxd5nh-fluxbox-1.3.7/bin/fluxbox "
```

---

```suggestion
    xsession.windowManager.command = concatStringsSep " " [ 
       "${cfg.package}/bin/fluxbox" 
       (escapeShellArgs (remove "" cfg.extraCommandLineArgs))
     ];
```

---

I think this can be simplified further. Please give

```suggestion
    xsession.windowManager.command = escapeShellArgs (
      [ "${cfg.package}/bin/fluxbox" ]
      ++ remove "" cfg.extraCommandLineArgs);
```

a try.